cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -683,55 +774,76 @@ public void checkCustomCaSecret(CertificateAuthority ca, Secret certSecret, Secr
             }
         }
 
-        Future<ReconciliationState> kafkaUpgrade() {
-            String kafkaStsName = KafkaCluster.kafkaClusterName(name);
-            return kafkaSetOperations.getAsync(namespace, kafkaStsName).compose(
-                sts -> {
-                    if (sts == null) {
+        private KafkaVersionChange getKafkaVersionChange(StatefulSet kafkaSts) {
```

Comment:
Can you explain what exactly did you changed here and why? Did you dropped the special handling for older Strimzi versions?

Reply:
The `getKafkaVersionChange` logic used to live in the `KafkaUpgrade` method, but the new ZK version change method needs to know those details too so I moved that code into its own method. 

I am not sure what you mean about code for handling older strimzi versions?